packages/core/src/devtools/utils.ts

```diff
@@ -21,11 +22,30 @@ export function getDebugClassNames(
       const matching = parentDebugClassNames.find(({ className: parentClassName }) => parentClassName === className);
 
       if (!matching && parentLookupItem[0][propertyHash]) {
-        // style is overriden in current merging, by rules in parent debug node
+        // parent node does not have current className (style), but has current selector:
+        // style is overriden in current merging by another rule in sibling node
         overriddenBy = getDirectionalClassName(parentLookupItem[0][propertyHash], parentLookupItem[1]);
-      } else {
-        // overridden status come from higher level of merging
-        overriddenBy = matching?.overriddenBy;
+      } else if (matching && parentLookupItem[0][propertyHash]) {
+        // parent node has current className (style), and has current selector:
+        // case 1. style is not overriden during current merging; it may be overriden in higher level of merging
+        // case 2. style is overriden in current merging by exactly the same rule in sibling nodes
+        const siblingHasSameRule = overridingSiblings?.filter(
+          overridingSibling =>
+            overridingSibling.debugClassNames.filter(
+              ({ className: siblingClassName }) => siblingClassName === className,
+            ).length,
+        ).length;
```

Comment:
```suggestion
            ).length > 0,
        ).length > 0;
```

_nit_ Let's makes "booleans" booleans 🐱